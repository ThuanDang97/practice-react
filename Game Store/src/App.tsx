import '../styles/reset.css'
import '../styles/globals.css'
import { Outlet, Route, Routes } from 'react-router-dom'
import { ThemeProvider } from '@mui/material/styles'
import { SWRConfig } from 'swr'
// Themes
import theme from '@self-types/Themes'

//page
import LoginPage from '@pages/login'
import ListPage from '@pages/index'
import DetailPage from '@pages/detail'
import RegisterPage from '@pages/register'

// Layout
import Header from '@layouts/Header'

// Contexts
import { AuthProvider } from '@contexts/AuthProvider'
import { CartProvider } from '@contexts/CartProvider'
import { swrFetcher } from './helpers'

const Dashboard = () => {
  return (
    <>
      <Header />
      <Outlet />
    </>
  )
}

const App = () => {
  return (
    <ThemeProvider theme={theme}>
      <SWRConfig
        value={{
          fetcher: swrFetcher,
        }}
      >
        <AuthProvider>
          <CartProvider>
            <Routes>
              <Route path="/" element={<Dashboard />}>
                <Route index element={<ListPage />} />
                <Route path="/games/:id" element={<DetailPage />} />
                <Route path="/games/search=:name" element={<ListPage />} />
              </Route>
              <Route path="/login" element={<LoginPage />} />
              <Route path="/register" element={<RegisterPage />} />
            </Routes>
          </CartProvider>
        </AuthProvider>
      </SWRConfig>
    </ThemeProvider>
  )
}

export default App
