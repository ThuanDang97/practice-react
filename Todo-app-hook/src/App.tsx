import './styles/reset.css';
import './styles/main.css';
import Todos from './pages/Todos';

const App = (): JSX.Element => {
  return <Todos />;
};
export default App;
