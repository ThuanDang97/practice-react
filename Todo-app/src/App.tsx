import './styles/reset.css';
import './styles/main.css';
import Header from './layouts/Header';
import React, { Component } from 'react';
import Footer from './layouts/Footer';
import TodoList, { ITodo } from './component/TodoList';
import { todos } from './constants/MockData';

class App extends Component<void> {
  state: {
    todos: ITodo[];
    filterValue: string;
  };
  constructor() {
    super();
    this.state = { todos, filterValue: 'All' };
  }

  handleAddTodo = (text: string) => {
    this.setState({
      todos: [...this.state.todos, { id: Date.now(), text, status: 'Active' }],
    });
  };

  handleFilterTodo = (value: string) => {
    this.setState({ filterValue: value });
  };

  render(): JSX.Element {
    const { todos, filterValue } = this.state;
    const filteredTodo =
      filterValue === 'All' ? todos : todos.filter((todo) => todo.status === filterValue);
    return (
      <div className="todoapp">
        <Header addTodo={this.handleAddTodo} />
        <TodoList listTodos={filteredTodo} />
        <Footer filterTodo={this.handleFilterTodo} lengthTodo={filteredTodo.length} />
      </div>
    );
  }
}

export default App;
