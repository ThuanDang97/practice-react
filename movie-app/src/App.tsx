import { Route, Routes } from 'react-router-dom';
import { Header } from './components/Header/Header';
import ListMoviePage from './pages/ListMoviePage';
import { NotFoundPage } from './pages/NotFoundPage';
import SingleMoviePage from './pages/SingleMoviePage';
import Footer from './components/Footer/Footer';

const App = () => {
  return (
    <>
      <Header />
      <Routes>
        <Route path="/" element={<ListMoviePage />}>
          <Route path=":movieType" element={<ListMoviePage />}></Route>
        </Route>
        <Route path="movie/:id" element={<SingleMoviePage />}></Route>
        <Route path="*" element={<NotFoundPage />}></Route>
      </Routes>
      <Footer />
    </>
  );
};

export default App;
