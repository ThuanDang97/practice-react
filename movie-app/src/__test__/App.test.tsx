import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import App from '../App';

describe('App component', () => {
  test('App should matches DOM Snapshot', () => {
    const app = (
      <Router>
        <App />
      </Router>
    );

    const { container } = render(app);

    expect(container).toMatchSnapshot();
  });
});
