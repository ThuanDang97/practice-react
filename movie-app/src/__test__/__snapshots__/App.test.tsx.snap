// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component App should matches DOM Snapshot 1`] = `
<div>
  <header>
    <a
      href="/"
    >
      <h2
        class="logo"
      >
        <img
          alt=""
          src="IMAGE_MOCK"
        />
      </h2>
    </a>
    <nav
      data-testid="navbar"
    >
      <ul
        class="dropdown-menu"
      >
        <li
          class="dropdown-item"
        >
          <a
            href="/"
          >
            <span>
              Movies
            </span>
          </a>
          <div
            class="dropdown-content"
          >
            <a
              class="nav-link"
              data-testid="navitem"
              href="/"
            >
              <span>
                Popular
              </span>
            </a>
            <a
              class="nav-link"
              data-testid="navitem"
              href="/now-playing"
            >
              <span>
                Now Playing
              </span>
            </a>
            <a
              class="nav-link"
              data-testid="navitem"
              href="/up-coming"
            >
              <span>
                Upcoming
              </span>
            </a>
            <a
              class="nav-link"
              data-testid="navitem"
              href="/top-rated"
            >
              <span>
                Top Rated
              </span>
            </a>
          </div>
        </li>
        <li
          class="dropdown-item"
        >
          <a
            href="/favorite-movies"
          >
            <span>
              Favorite Movies
            </span>
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <div
    class="media"
    data-testid="home-page"
  >
    <div
      class="column_wrapper"
    >
      <div
        class="content_wrapper"
      >
        <div
          class="title"
        >
          <h2
            data-testid="header-page"
          >
            Popular Movies
          </h2>
        </div>
        <div
          class="content"
        >
          <div>
            <div
              class="panel-card"
            >
              <div
                class="name"
              >
                <h2>
                  Sort
                </h2>
              </div>
              <div
                class="filter"
              >
                <span
                  class="widget"
                  data-testid="sub-menu"
                >
                  <span
                    class="dropdown"
                    data-testid="select"
                  >
                    <span
                      class="input"
                      data-testid="titleOption"
                    >
                      Popularity Descending
                    </span>
                    <span
                      class="select"
                    >
                      <span
                        class="icon-select"
                      />
                    </span>
                  </span>
                </span>
              </div>
            </div>
            <div
              class="panel-card"
            >
              <div
                class="name"
              >
                <h2>
                  Filter
                </h2>
              </div>
              <div
                class="filter"
              >
                <span
                  class="widget"
                  data-testid="sub-menu"
                >
                  <div
                    class="list-chip"
                    data-testid="list-chip"
                  >
                    <h3>
                      Genres
                    </h3>
                  </div>
                  <div
                    class="ranger-slider"
                  >
                    <h3>
                      User Score
                    </h3>
                    <input
                      class="range"
                      max="10"
                      min="0"
                      step="1"
                      type="range"
                      value="10"
                    />
                    <div
                      class="range-value"
                    >
                      10
                    </div>
                  </div>
                </span>
              </div>
            </div>
            <div
              class="btn-layout"
            >
              <button
                class="btn primary contained rd-2 w-100 "
                type="button"
              >
                Search
              </button>
            </div>
          </div>
          <div
            class="white_column"
          >
            <section
              class="page_wrapper"
            >
              <span
                data-testid="message-error"
              >
                No items were found that match your query.
              </span>
            </section>
            <div
              class="btn-layout"
            >
              <button
                class="btn primary contained rd-1 w-100 "
                disabled=""
                type="button"
              >
                Load More
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <footer>
    <a
      href="/"
    >
      <h2
        class="logo_footer"
      >
        <img
          alt=""
          src="IMAGE_MOCK"
        />
      </h2>
    </a>
    <div
      class="menu_footer"
    >
      <div
        class="nav_footer"
      >
        <h3
          class="nav_title"
        >
          THE BASICS
        </h3>
        <div
          class="sub_nav"
        >
          <a
            class="sub_nav_item"
            href="/"
          >
            <span>
              About TMDB
            </span>
          </a>
          <a
            class="sub_nav_item"
            href="/"
          >
            <span>
              Contact Us
            </span>
          </a>
          <a
            class="sub_nav_item"
            href="/"
          >
            <span>
              Support Forums
            </span>
          </a>
          <a
            class="sub_nav_item"
            href="/"
          >
            <span>
              API
            </span>
          </a>
          <a
            class="sub_nav_item"
            href="/"
          >
            <span>
              System Status
            </span>
          </a>
        </div>
      </div>
      <div
        class="nav_footer"
      >
        <h3
          class="nav_title"
        >
          GET INVOLVED
        </h3>
        <div
          class="sub_nav"
        >
          <a
            class="sub_nav_item"
            href="/"
          >
            <span>
              Contribution Bible
            </span>
          </a>
          <a
            class="sub_nav_item"
            href="/"
          >
            <span>
              Add New Movie
            </span>
          </a>
          <a
            class="sub_nav_item"
            href="/"
          >
            <span>
              Add New TV Show
            </span>
          </a>
        </div>
      </div>
      <div
        class="nav_footer"
      >
        <h3
          class="nav_title"
        >
          COMMUNITY
        </h3>
        <div
          class="sub_nav"
        >
          <a
            class="sub_nav_item"
            href="/"
          >
            <span>
              Guidelines
            </span>
          </a>
          <a
            class="sub_nav_item"
            href="/"
          >
            <span>
              Discussions
            </span>
          </a>
          <a
            class="sub_nav_item"
            href="/"
          >
            <span>
              Leaderboard
            </span>
          </a>
          <a
            class="sub_nav_item"
            href="/"
          >
            <span>
              Twitter
            </span>
          </a>
        </div>
      </div>
      <div
        class="nav_footer"
      >
        <h3
          class="nav_title"
        >
          LEGAL
        </h3>
        <div
          class="sub_nav"
        >
          <a
            class="sub_nav_item"
            href="/"
          >
            <span>
              Terms of Use
            </span>
          </a>
          <a
            class="sub_nav_item"
            href="/"
          >
            <span>
              API Terms of Use
            </span>
          </a>
          <a
            class="sub_nav_item"
            href="/"
          >
            <span>
              Privacy Policy
            </span>
          </a>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
